tree/dataframe/src/RLoopManager.cxx

```diff
@@ -474,9 +541,23 @@ void RLoopManager::RunTreeProcessorMT()
 /// Run event loop over one or multiple ROOT files, in sequence.
 void RLoopManager::RunTreeReader()
 {
+   if (fStartEntry == fEndEntry) // empty range => no work needed
+      return;
+
    TTreeReader r(fTree.get(), fTree->GetEntryList());
+
    if (0 == fTree->GetEntriesFast())
       return;
```

Comment:
can you please merge this check with the one above?

Reply:
uh?  where do you think TTreeReader gets that info from? :smile: but anyway `GetEntriesFast` does not trigger any reading, it just checks whether `fTree` already knows that it has 0 entries.

To clarify my suggestion is to change 

```cpp
   if (fStartEntry == fEndEntry) // empty range => no work needed
      return;

   TTreeReader r(fTree.get(), fTree->GetEntryList());

   if (0 == fTree->GetEntriesFast())
      return;
```

to

```cpp
   if (fStartEntry == fEndEntry || 0 == fTree->GetEntriesFast()) 
      return; // nothing to do

   TTreeReader r(fTree.get(), fTree->GetEntryList());
```

Reply:
I think the cleanest solution is:
```cpp
void RLoopManager::RunTreeReader()
{
   TTreeReader r(fTree.get(), fTree->GetEntryList());

   const auto nEntries = r.GetEntries(/*force=*/ true);

   if (fStartEntry == fEndEntry || 0 == nEntries)
      return;

   // if `GetEntries` is not called, in case of chains, `SetEntriesRange` might not properly indicate that a
   // start entry is too large, e.g. when start=last entry
   // the `GetEntries` call can be removed once `https://github.com/root-project/root/issues/10774` is resolved
   if (fStartEntry >= nEntries)
      throw std::logic_error(); // start is too big
   r.SetEntriesRange(fStartEntry, fEndEntry);
```
Basically leave the TTreeReader do all the work obtaining the entries certainly for once. (And after that SetEntriesRange)

Reply:
Although I like the concept in general, `GetEntries(true)` means opening all the files to count all the entries. Doing it outside of the TTreeReader means that we're for sure wasting the possibility of keeping the files open. Probably we can find a way to open all files only once for the whole run